docs/maintainers/release.md

```diff
@@ -1,44 +1,78 @@
 # Antrea Release Process
 
 This file documents the list of steps to perform to create a new Antrea
-release. We use `<TAG>` as a placeholder for the release tag (e.g. `v0.1.0`).
-
-* Create a release branch for each new minor release (e.g `release-0.1`). For
-  bug fixes, use the appropriate release branch.
-
-* Open a PR against the appropriate release branch with the following commits:
-  1. a commit to update the [CHANGELOG](../../CHANGELOG). All significant
-     changes and all bug fixes since the first version of the previous minor
-     release should be mentioned. The commit message must be *exactly* `"Update
-     CHANGELOG for <TAG> release"`, as a bot will look for this commit and
-     cherry-pick it to update the main branch (starting with Antrea v1.0). The
-     [process-changelog.go](../../hack/release/process-changelog.go) script
-     should be used to easily generate links to PRs and the Github profiles of
-     PR authors.
-  2. a commit to update [VERSION](../../VERSION) as needed.
-
-* Make the release on Github with the release branch as the target: copy the
-  relevant section of the [CHANGELOG](../../CHANGELOG) for the release
-  description and check the `pre-release` box if applicable. There is no need to
-  upload any assets as this will be done automatically by a Github workflow,
-  after you create the release.
-
-* After a while (time for the Github workflows to complete), check that:
-  1. the docker image has been pushed to
+release. We use `<TAG>` as a placeholder for the release tag (e.g. `v1.4.0`).
+
+1. *For a minor release* On the code freeze date (typically one week before the
```

Comment:
shall we end this bullet point with what needs to be done differently for non-minor releases (esp patch release)? or say that "skip this step for patch releases"

Reply:
I feel like it is pretty clear as it is, but I may be biased